visual-tests/test-scripts/series/stacked-bar.js

```diff
@@ -9,17 +9,38 @@
 
     d3.csv('stackedBarData.csv', function(error, data) {
 
-        var categories = data.slice(1).map(function(d) { return d.State; });
-        var yMax = d3.max(data, function(d) {
-            var total = 0;
-            for (var prop in d) {
-                if (prop !== 'State') {
-                    total += Number(d[prop]);
+        var categories = data.map(function(d) { return d.State; });
+
+        // Format data into a map of series key to series data & a data array to pass to our stacked-bar series.
+        var seriesMap = {};
+        var seriesData = [];
+        data.forEach(function(stateData) {
+            for (var propertyName in stateData) {
+                if (stateData.hasOwnProperty(propertyName) && propertyName !== 'State') {
+
+                    if (!seriesMap[propertyName]) {
+                        var dataArray = [];
+                        // Store the series names mapped to the data arrays.
+                        seriesMap[propertyName] = dataArray;
+                        // Store the data arrays to pass to our stackedBar series.
+                        seriesData.push(dataArray);
+                    }
+
+                    seriesMap[propertyName].push({
+                        x: stateData.State,
+                        y: parseInt(stateData[propertyName])
+                    });
                 }
             }
-            return total;
         });
 
+        var stack = fc.series.stackedBar();
+
+        // Manually compute stacked data - use this to calculate the max Y domain.
+        stack.stackedData(stack.layout.value(seriesData));
```

Comment:
Oooohhhhhhh, nice one. This is a good example of why you should get other people to try out your code! It hadn't occurred to me that this would be a consequence of our approach to properties, we should definitely revisit this separately (#275).

Our normal pattern is to use the d3 accessor i.e. ```stack.layout()(seriesData)```.